subprojects/chrome-trace/src/main/java/org/gradle/trace/listener/BuildOperationListenerAdapter.java

```diff
@@ -18,7 +18,7 @@ static BuildOperationListenerAdapter create(Gradle gradle, TraceResult traceResu
         if (gradleVersion.matches("4\\.[0-6](\\D.*)?")) {
             return new Gradle40BuildOperationListenerAdapter(gradle, new Gradle40BuildOperationListenerInvocationHandler(traceResult));
         }
-        if (gradleVersion.startsWith("4.")) {
+        if (gradleVersion.startsWith("4.") || gradleVersion.startsWith("5.")) {
```

Comment:
This sounds flimsy. Shouldn't this be `> 4.0`? I.e. add a `VersionNumber` type that can parse and compare version numbers and use that?